owntracks.go

```diff
@@ -30,23 +37,89 @@ func (o *owntracksServer) HandlePublish(w http.ResponseWriter, r *http.Request)
 
 	// parse message
 	msg := owntracksMessage{}
-	if err := json.NewDecoder(r.Body).Decode(&msg); err != nil {
+	rawMsg, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		o.log.Printf("read owntracks message: %v", err)
+		http.Error(w, fmt.Sprintf("read owntracks message: %v", err), http.StatusInternalServerError)
+		return
+	}
+	if err := json.Unmarshal(rawMsg, &msg); err != nil {
 		o.log.Printf("decoding owntracks message: %v", err)
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
 	// ignore if not location
-	if msg.IsLocation() {
-		if err := o.store.AddOTLocation(r.Context(), msg); err != nil {
-			o.log.Printf("saving owntracks location: %v", err)
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-			return
+	if !msg.IsLocation() {
+		o.log.Printf("ignoring payload type %s", msg.Type)
+		w.Header().Set("Content-Type", "application/json")
+		fmt.Fprint(w, `[]`)
+		return
+	}
+
+	var errs []string
+
+	if err := o.store.AddOTLocation(r.Context(), msg); err != nil {
+		errs = append(errs, fmt.Sprintf("saving owntracks location: %v", err))
+	}
+
+	if o.recorderURL != nil {
+		if err := o.proxyLocation(r, rawMsg); err != nil {
```

Comment:
Is there any point in sending the other messages to recorder? Does it even handle them, or is this “good enough”?

Reply:
This only handles `_type=location` payloads:

https://github.com/lstoll/wherewasi/pull/16/files/3fb6ffbcee71d030cf8c85617993af41e8a0b9cd#diff-fe0fc9ba77cd1a3dd82a96bbb832b00aR53

How would you handle other types of payload? Just pass-through?